app/src/main/java/com/wednesday/bankingdetails/adapter/BankListAdapter.kt

```diff
@@ -25,6 +29,35 @@ class BankListAdapter(var bankDataSource: ArrayList<Bank>) :
             itemView.txtIfsCode.text = ifsc.toUpperCase(
                 Locale.getDefault()
             )
+            if (bank.isFavourite) {
+                itemView.imgStar.visibility = View.VISIBLE
+            } else {
+                itemView.imgStar.visibility = View.GONE
+            }
+
+
+            val bankDatabase = FavouritesDatabase.getDatabase(itemView.context)
```

Comment:
This does not make sense. The only job of the adapter is to take data and bind it to the views. When you reference the database here you are create a hard coupling. Now this adapter can only be used with the database. Create an interface and give a callback to the object using it. The call back should have:

- Reference to the bank

The owner of the datasource i.e. activity will then update the datasource, save it in the db and call reset on the adapter.

Reply:
Did it using BankFavouriteListener